### ISSUE TITLE
list() to [] with optional args don't keep the last comma

### ISSUE THREAD

**Issue by carusogabriel** [slevomat/coding-standard#417]
```diff
-list(, $foo, ) = $this->bar->method($baz);
+[, $parent ] = $this->bar->method($baz);
```

**Comment by Levivb** [comment]
Not sure, but does it matter?

I've tinkered a bit, but functionality seems untouched:
```php
>>> [, $x, ] = (function(){ return [1,2,3]; })();
>>> $x // => 2
```
Is similar to:
```php
>>> [, $x] = (function(){ return [1,2,3]; })();
>>> $x // => 2
```

One reason I could come up with to keep the trailing comma is to make the intent clear. In which you explicitly add a trailing comma to show the method returns an array of length 3.

**Comment by carusogabriel** [comment]
> Not sure, but does it matter?

Nops, but we shouldn't do it, Sniffs should only change what it is supposed to.

> Are you really sure it’s broken by the `ShortListSniff`?

@kukulich I'll try to isolate an example to reproduce it.

**Comment by kukulich** [comment]
@carusogabriel Thanks. The `ShortListSniff` is so simple that I really think it is innocent :)

**Comment by Majkl578** [comment]
This isn't caused by ShortListSniff, it only replaces the `list(` and `)` parts, nothing else.

This is a bug in Squiz.Arrays.ArrayDeclaration:
```
3 | ERROR | [x] Comma not allowed after last value in single-line array declaration (Squiz.Arrays.ArrayDeclaration.CommaAfterLast)
```
for:
```php
<?php

[, $x, ] = [1, 2, 3];
```

**Comment by Majkl578** [comment]
@carusogabriel Please report this to [PHP_CodeSniffer](https://github.com/squizlabs/PHP_CodeSniffer/) directly, thanks. :)
